File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int pos = m + n - 1;
        int i = m-1, j = n-1;
        while (pos >= 0){
            if (i < 0){
                nums1[pos] = nums2[j];
                j--;
                pos--;
                continue;
            }
            if (j < 0){
                nums1[pos] = nums1[i];
                i--;
                pos--;
                continue;
            }
            if (nums1[i] >= nums2[j]){
                nums1[pos] = nums1[i];
                i--;
            } else {
                nums1[pos] = nums2[j];
                j--;
            }
            pos--;
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
    sorted := make([]int, m+n)
    i, j, k := 0, 0, 0
    for i < m || j < n {
        first, second := math.MaxInt, math.MaxInt
        if i < m {
            first = nums1[i]
        }
        if j < n {
            second = nums2[j]
        }

        if first <= second {
            sorted[k] = first
            i++
        } else {
            sorted[k] = second
            j++
        }
        k++
    }
    copy(nums1, sorted)
}